engine/src/cubos/engine/imgui/data_inspector.cpp

```diff
@@ -0,0 +1,309 @@
+            }
+
+            if (!readOnly)
+            {
+                // Adding key
+                ImGui::TableNextRow();
+                ImGui::TableSetColumnIndex(0);
+                ImGui::AlignTextToFramePadding();
+                ImGui::PushID("edit_dict_item");
+
+                if (ImGui::Button("Add item"))
+                {
+                    mKeyExists = false;
+                    mKey = AnyValue::defaultConstruct(trait.keyType());
+                    mValue = AnyValue::defaultConstruct(trait.valueType());
+                    ImGui::OpenPopup("DictPopup");
+                }
+
+                ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+                ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5F, 0.5F));
+
+                if (ImGui::BeginPopupModal("DictPopup", nullptr, ImGuiWindowFlags_AlwaysAutoResize))
+                {
+                    ImGui::BeginTable("DictPopupInsert", 2);
+                    inspect("key", trait.keyType(), mKey.get(), readOnly);
+                    inspect("value", trait.valueType(), mValue.get(), readOnly);
+                    ImGui::EndTable();
+
+                    if (mKeyExists)
+                    {
+                        ImGui::TextColored(ImVec4(1.0F, 0.0F, 0.0F, 1.00F), "Key already exists.");
+                    }
+
+                    if (ImGui::Button("Add", ImVec2(80, 0)))
+                    {
+                        if (view.find(mKey.get()) != view.end())
+                        {
+                            mKeyExists = true;
+                        }
+                        else
+                        {
+                            mKeyExists = false;
+                            view.insertMove(mKey.get(), mValue.get());
+                            changed = true;
+                            ImGui::CloseCurrentPopup();
+                        }
+                    }
+
+                    ImGui::Separator();
+
+                    if (ImGui::Button("Cancel", ImVec2(80, 0)))
+                    {
+                        mKeyExists = false;
+                        ImGui::CloseCurrentPopup();
+                    }
+
+                    ImGui::EndPopup();
+                }
+
+                ImGui::PopID();
+            }
+        });
+    }
+    else if (type.has<FieldsTrait>())
+    {
+        const auto& trait = type.get<FieldsTrait>();
+        const auto& view = trait.view(value);
+        if (trait.size() == 1)
+        {
+            auto [field, fieldVal] = *view.begin();
+            changed |= inspect(name, field->type(), fieldVal, readOnly);
+        }
+        else
+        {
+            showStructured(name, type, [&]() {
+                for (auto [field, fieldVal] : view)
+                {
+                    changed |= inspect(field->name(), field->type(), fieldVal, readOnly);
+                }
+            });
+        }
+    }
+    else
```

Comment:
I think you need to handle `StringConversionTrait` here: for example, `RenderableGrid` components contain a `Asset<Grid>` field which only provides that trait.